scikeras/wrappers.py

```diff
@@ -871,192 +874,218 @@ def preprocess_y(y):
+                else:
+                    encoder = make_pipeline(
+                        Ensure2DTransformer(), OrdinalEncoder(dtype=np.float32),
+                    )
+            else:
+                encoder = self.encoders_[0]
             y = encoder.fit_transform(y)
-            classes_ = encoder.classes_
+            classes_ = encoder[1].categories_[0]
+            n_classes_ = len(classes_)
             # make lists
             encoders_ = [encoder]
             classes_ = [classes_]
-            y = [y]
-        elif target_type_ == "multilabel-indicator":
-            # y = array([1, 1, 1, 0], [0, 0, 1, 1])
-            # split into lists for multi-output Keras
-            # will be processed as multiple binary classifications
-            classes_ = [np.array([0, 1])] * y.shape[1]
-            y = np.split(y, y.shape[1], axis=1)
-            model_n_outputs_ = len(y)
-            # encode
-            encoders_ = [LabelEncoder() for _ in range(len(y))]
-            y = [
-                encoder.fit_transform(y_.reshape(-1,) if y_.shape[1] == 1 else y_)
-                for encoder, y_ in zip(encoders_, y)
-            ]
-            classes_ = [encoder.classes_ for encoder in encoders_]
-        elif target_type_ == "multiclass-multioutput":
-            # y = array([1, 0, 5], [2, 1, 3])
-            # split into lists for multi-output Keras
-            # each will be processesed as a seperate multiclass problem
-            y = np.split(y, y.shape[1], axis=1)
-            model_n_outputs_ = len(y)
+            n_classes_ = [n_classes_]
+        elif self.target_type_ == "multiclass-one-hot":
+            # y = array([1, 0], [1, 0]) (usually used with categorical_crossentropy)
+            model_n_outputs_ = 1  # single softmax output expected
+            n_outputs_ = 1
             # encode
-            encoders_ = [LabelEncoder() for _ in range(len(y))]
-            y = [
-                encoder.fit_transform(y_.reshape(-1,) if y_.shape[1] == 1 else y_)
-                for encoder, y_ in zip(encoders_, y)
-            ]
-            classes_ = [encoder.classes_ for encoder in encoders_]
+            # here we just pass through to allow all Keras use cases
+            # that didn't exist in sklearn to succeed (ex: loss=categorical_crossentropy)
+            if reset:
+                encoder = FunctionTransformer()
+            else:
+                encoder = self.encoders_[0]
+            y = encoder.fit_transform(y)
+            # there is no real "classes" representation for this type of data
+            classes_ = None
+            n_classes_ = y.shape[1]
+            # make lists
+            encoders_ = [encoder]
+            classes_ = [classes_]
+            n_classes_ = [n_classes_]
         else:
-            raise ValueError("Unknown label type: {}".format(target_type_))
+            raise ValueError(
+                f"Unknown label type: {self.target_type_}."  # To match errors used by sklearn
```

Comment:
This error gets triggered if the user has `target_type = "multiclass-indicator"` but not `target_type = "multiclass-one-hot"`, right?

Reply:
Right. `"multiclass-one-hot"` is a normal use case in Keras so I feel that it should work out of the box with SciKeras (using the same Keras model).

Reply:
That means this error is raised if the user passes data that yields `target_type = "multiclass-indicator"`. That sounds like it'd be a bug report.

Reply:
Because, as far as I can tell, there is no obvious thing to map to in Keras, right? For any multi-output model (`"multiclass-indicator"` or `"multiclass-multioutput"`) you could have multiple combinations of outputs.

Reply:
Closing this thread in favor of https://github.com/adriangb/scikeras/pull/88/files#r499172803